app/src/main/java/com/babylon/wallet/android/domain/usecases/BuildDappResponseUseCase.kt

```diff
@@ -183,7 +183,7 @@ class BuildAuthorizedDappResponseUseCase @Inject constructor(
                 )
                 val signRequest = SignRequest.SignAuthChallengeRequest(
                     challengeHex = authRequest.challenge.hex,
-                    origin = request.metadata.origin,
+                    origin = request.metadata.origin.substring(0, request.metadata.origin.length - 1),
```

Comment:
maybe `.removeTrailingSlash()`? Or if it should remove any last character and this code is not temporary it's worth checking that endIndex is larger than the length of the String

Reply:
this is testing leftover, indeed there is method with name almost exactly matching your suggestion for this. Good spot!